tensorflow/core/kernels/generate_box_proposals_op.cu.cc

```diff
@@ -99,58 +109,47 @@ __global__ void GeneratePreNMSUprightBoxesKernel(
 
       // TODO use fast math when possible
 
-      // Deltas of shape (N,H,W,A4)
-      int deltas_idx = box_conv_index + image_index * KA;
+      // Deltas of shape (N,height,width,A4)
+      int deltas_idx = box_conv_index + image_index * image_stride;
       float4 deltas = d_bbox_deltas[deltas_idx];
       float dx = deltas.y;
       float dy = deltas.x;
       float dw = deltas.w;
       float dh = deltas.z;
-      // printf("deltas_idx=%d dx=%f, dy=%f, dw=%f,
-      // dh=%f\n",deltas_idx,dx,dy,dw,dh);
       // Upper bound on dw,dh
       dw = fmin(dw, bbox_xform_clip);
       dh = fmin(dh, bbox_xform_clip);
 
       // Applying the deltas
-      float width = x2 - x1 + 1.0f;
+      float width = legacy_op<T>(x2 - x1);
       const float ctr_x = x1 + 0.5f * width;
       const float pred_ctr_x = ctr_x + width * dx;  // TODO fuse madd
       const float pred_w = width * expf(dw);
       x1 = pred_ctr_x - 0.5f * pred_w;
-      x2 = pred_ctr_x + 0.5f * pred_w;
+      x2 = -legacy_op<T>(-(pred_ctr_x + 0.5f * pred_w));
 
-      float height = y2 - y1 + 1.0f;
+      float height = legacy_op<T>(y2 - y1);
       const float ctr_y = y1 + 0.5f * height;
       const float pred_ctr_y = ctr_y + height * dy;
       const float pred_h = height * expf(dh);
       y1 = pred_ctr_y - 0.5f * pred_h;
-      y2 = pred_ctr_y + 0.5f * pred_h;
+      y2 = -legacy_op<T>(-(pred_ctr_y + 0.5f * pred_h));  // -1 if legacy_op
 
-      if (correct_transform) {
-        x2 -= 1.0f;
-        y2 -= 1.0f;
-      }
-      // const float y2_old=y2;
-      // const float x2_old=x2;
-      // const float x1_old=x1;
-      // const float y1_old=y1;
       // Clipping box to image
       const float img_height = d_img_info_vec[5 * image_index + 0];
       const float img_width = d_img_info_vec[5 * image_index + 1];
       const float min_size_scaled =
           min_size * d_img_info_vec[5 * image_index + 2];
-      // min_size * d_img_info_vec[3 * image_index + 2];
-      x1 = fmax(fmin(x1, img_width - 1.0f), 0.0f);
-      y1 = fmax(fmin(y1, img_height - 1.0f), 0.0f);
-      x2 = fmax(fmin(x2, img_width - 1.0f), 0.0f);
-      y2 = fmax(fmin(y2, img_height - 1.0f), 0.0f);
+      x1 = fmax(fmin(x1, -legacy_op<T>(-img_width)), 0.0f);
```

Comment:
Applying minus sign twice makes it harder to read, maybe add an `SubtractOffset` func for it.

Reply:
Done

Reply:
Done